glotaran/analysis/scheme.py

```diff
@@ -85,7 +87,22 @@ def from_yml_file(cls, filename: str) -> 'Scheme':
         nnls = scheme.get('nnls', False)
         nfev = scheme.get('nfev', None)
         group_tolerance = scheme.get('group_tolerance', 0.0)
-        return cls(model=model, parameter=parameter, data=data,
+
+        extra = scheme.get('extra', {})
+
+        for label, path in extra.items():
+            path = pathlib.Path(path)
+
+            fmt = path.suffix[1:] if path.suffix != '' else 'nc'
+            if 'data_format' in scheme:
```

Comment:
The thing is: if you define `data_format` in the scheme then ALL data is assumed to be in that format, otherwise it will deduced from the suffix (defaulting to `nc`). So having mixed data formats is possible, if the paths have the right suffix.